File: Códigos em C e C++/Ideias para orientação a objetos/produto.cpp
#include "produto.h";

Produto::Produto(){
    char nome[19];
    int ean = 0;
    float preco = 0.0;
    int quantidade = 0;
}

Produto::Produto (char n, int e, float p, int q){
    setProduto(char n, int e, float p, int q)
}

void Produto::setProduto(char n, int e, float p, int q){
    char nome[19] {n};
    int ean = e;
    float preco = p;
    int quantidade = q;
}
void impremeProduto(){

    for (int i = 0; i < 19; i++){
        cout << n[i];
    }
    cout << e;
    cout << p;
    cout << q;
}


~Produto()nome[];

File: C#/SextAula/SextaAula/Produto.cs
using System;


namespace SextaAula
{
    public class Produto
    {
        public string Nome;
        public double Preco;
        public int Quantidade;

        public double ValTotEst() 
        {
            return Preco*Quantidade;
        }

        public void AdcionarProduto(int quantidade) 
        {
            Quantidade+=quantidade;
        }

        public void RemoverProduto(int quantidade)
        {
            Quantidade-=quantidade;
        } 

        public override string ToString()
        {
            return $" Dados do produto: {Nome}, R$ {Preco} Quantidade {Quantidade}, Total: R$ {ValTotEst().ToString("F2")}";
        }
    }
}